tsdb/head_append.go

```diff
@@ -978,23 +978,39 @@ func exemplarsForEncoding(es []exemplarWithSeriesRef) []record.RefExemplar {
 	return ret
 }
 
-// Commit writes to the WAL and adds the data to the Head.
-// TODO(codesome): Refactor this method to reduce indentation and make it more readable.
-func (a *headAppender) Commit() (err error) {
-	if a.closed {
-		return ErrAppenderClosed
-	}
-	defer func() { a.closed = true }()
-
-	if err := a.log(); err != nil {
-		_ = a.Rollback() // Most likely the same error will happen again.
-		return fmt.Errorf("write to WAL: %w", err)
-	}
-
-	if a.head.writeNotified != nil {
-		a.head.writeNotified.Notify()
-	}
+type appenderCommitContext struct {
+	floatsAppended     int
+	histogramsAppended int
+	// Number of samples out of order but accepted: with ooo enabled and within time window.
+	oooFloatsAccepted    int
+	oooHistogramAccepted int
+	// Number of samples rejected due to: out of order but OOO support disabled.
+	floatOOORejected int
+	histoOOORejected int
+	// Number of samples rejected due to: out of order but too old (OOO support enabled, but outside time window).
+	floatTooOldRejected int
+	histoTooOldRejected int
+	// Number of samples rejected due to: out of bounds: with t < minValidTime (OOO support disabled).
+	floatOOBRejected    int
+	histoOOBRejected    int
+	inOrderMint         int64
+	inOrderMaxt         int64
+	oooMinT             int64
+	oooMaxT             int64
+	wblSamples          []record.RefSample
+	wblHistograms       []record.RefHistogramSample
+	wblFloatHistograms  []record.RefFloatHistogramSample
+	oooMmapMarkers      map[chunks.HeadSeriesRef][]chunks.ChunkDiskMapperRef
+	oooMmapMarkersCount int
+	oooRecords          [][]byte
+	oooCapMax           int64
+	series              *memSeries
```

Comment:
Turns out you don't need `series` here, you can just created it in the relevant functions with `:=`. For example in `commitSamples`:
```
series := a.sampleSeries[i]
```

Same in the other commit functions, in fact `commitMetadata` won't need `acc` input.